Make bookmarking possible with multiple cursors

solendil opened the pull request:
There was an attempt in the code to implement multiple cursors (related to #59)

```coffee
for cursor in cursors		
  range = @editor.getSelectedBufferRange()		 
  bookmarks = @markerLayer.findMarkers(intersectsBufferRowRange: [range.start.row, range.end.row])		       
```

I figured I could make it work, so here's the fix, along with its jasmine spec.

50Wliu commented:
Can you also add a spec for simultaneously adding and removing bookmarks?

50Wliu commented on lib/bookmarks.coffee:
```diff
@@ -32,8 +32,7 @@ class Bookmarks
 
   toggleBookmark: =>
     cursors = @editor.getCursors()
```
This looks like it's unused now.

50Wliu commented:
Interestingly, `intersectsBufferRowRange` doesn't seem to be a valid option.  The closest there is is `intersectsRowRange`.  https://github.com/atom/text-buffer/blob/v8.5.0/src/marker-layer.coffee#L94

solendil commented:
@50Wliu I removed the unused line of code and changed `intersectsBufferRowRange` to `intersectsRowRange`. How did it work? I'm not sure, but it still does.
 
As for the specs, there are 

    describe "multiple point marker bookmark", ->
        it "creates multiple markers when toggled", ->


and

     it "removes multiple markers when toggled", ->

Do you think more is needed?

50Wliu commented:
Yeah, I think it's best to make sure that adding and removing at the same time works as well.

solendil commented:
Here it is : this new spec tests the mechanism in some extreme scenario. It seems unlikely that anybody will ever try that in real-life, but if they do, it will work :)
